tests/transformers/test_transformers_model_export.py

```diff
@@ -898,3 +970,563 @@ def postprocess(self, model_outputs):
+
+    assert second_response == "A boat that can go to Antarctica."
+
+    # Test that a new loaded instance has no context.
+    loaded_again_pyfunc = mlflow.pyfunc.load_model(model_path)
+    third_response = loaded_again_pyfunc.predict("What kind of boat should I use?")
+
+    assert third_response == "A boat that can't sink."
+
+    fourth_response = loaded_again_pyfunc.predict("Can I use it to go to the moon?")
+
+    assert fourth_response == "Only if you have a boat that can't sink."
+
+
+@pytest.mark.parametrize(
+    "input, result",
+    [
+        (
+            "I have a dog and his name is Willy!",
+            "PRON,VERB,DET,NOUN,CCONJ,PRON,NOUN,AUX,PROPN,PUNCT",
+        ),
+        (
+            ["We are the knights who say nee!", "Houston, we may have a problem."],
+            [
+                "PRON,AUX,DET,PROPN,PRON,VERB,INTJ,PUNCT",
+                "PROPN,PUNCT,PRON,AUX,VERB,DET,NOUN,PUNCT",
+            ],
+        ),
+    ],
+)
+def test_ner_pipeline(ner_pipeline, model_path, input, result):
+    mlflow.transformers.save_model(ner_pipeline, model_path)
+    loaded_pyfunc = mlflow.pyfunc.load_model(model_path)
+    inference = loaded_pyfunc.predict(input)
+
+    assert inference == result
+
+
+@pytest.mark.parametrize(
+    "input, in_signature, out_signature",
+    [
+        ("Generative models are", '[{"type": "string"}]', '[{"type": "string"}]'),
+        (
+            ["Computers are", "Dogs are"],
+            '[{"type": "string"}]',
+            '[{"type": "string"}]',
+        ),
+    ],
+)
+def test_signature_inference_strings_and_lists_of_strings(
+    text_generation_pipeline, model_path, input, in_signature, out_signature
+):
+    pyfunc = mlflow.transformers._TransformersWrapper(text_generation_pipeline, None)
+
+    answer = pyfunc.predict(input)
+    signature = infer_signature(input, answer)
+
+    assert signature.inputs.to_json() == in_signature
+    assert signature.outputs.to_json() == out_signature
+
+
+@pytest.mark.parametrize(
+    "input, in_signature, out_signature",
+    [
+        (
+            {
+                "sequences": "This new MLflow flavor is awesome!",
+                "candidate_labels": '"happy", "sad"',
+            },
+            '[{"name": "sequences", "type": "string"}, {"name": "candidate_labels", '
+            '"type": "string"}]',
+            '[{"type": "string"}]',
+        ),
+    ],
+)
+def test_signature_inference_dict_input_to_string_output(
+    zero_shot_pipeline, input, in_signature, out_signature
+):
+    pyfunc = mlflow.transformers._TransformersWrapper(zero_shot_pipeline, None)
+
+    signature = infer_signature(input, pyfunc.predict(input))
+
+    assert signature.inputs.to_json() == in_signature
+    assert signature.outputs.to_json() == out_signature
+
+
+@pytest.mark.parametrize(
+    "input, in_signature, out_signature",
+    [
+        (
+            "question: Who's house? context: The house is owned by a man named Run.",
```

Comment:
On second thought, I think that Dict[str, str] is more natural for these types of models where multiple strings get formatted into a "prompt".

The string concat solution could break down if the question contains the string "context", for example.

Reply:
Changed this behavior to map to a Dict[str,str] and internally will convert to the required types for the Pipelines.